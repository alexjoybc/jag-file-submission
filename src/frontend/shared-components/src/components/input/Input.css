.label {
  display: flex;
  flex-wrap: wrap;
  align-items: baseline;
}

.textinput_label {
  font-weight: 700;
  font-style: normal;
  font-size: 16px;
  color: #333333;
  text-align: left;
  line-height: 16px;
}

.editable_white {
  padding: 1px 0px 1px 0px;
  box-sizing: border-box;
  top: 30px;
  width: 100%;
  height: 40px;
  background-color: white;
  font-weight: 400;
  font-style: normal;
  font-size: 16px;
  text-decoration: none;
  color: #494949;
  text-align: left;
  border-style: inset;
}

.note {
  font-size: 14px;
  font-weight: 400;
}

.error {
  color: red;
}

.mandatory {
  padding: 0px 0px 0px 3px;
  color: red;
  font-weight: 500;
  font-size: 16px;
}
