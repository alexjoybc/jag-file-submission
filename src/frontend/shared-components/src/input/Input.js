import React from "react";
import PropTypes from "prop-types";
import "./Input.css";

export const Input = ({
  input: { label, id, value, placeholder, isRequired, errorMsg, styling, note },
  onChange
}) => {
  let asterisk = "";

  if (isRequired) {
    asterisk = (
      <span id="asterisk" className="mandatory">
        *
      </span>
    );
  }

  let labelPart = null;
  const labelPartExists = label !== false;

  if (label) {
    labelPart = (
      <label htmlFor={id} className="label">
        <div className="textinput_label">{label}</div>
        {asterisk}&nbsp;
        <span className="note">{note}</span>
      </label>
    );
  }

  return (
    <div>
      {labelPartExists && labelPart}
      <input
        className={styling}
        type="text"
        id={id}
        defaultValue={value}
        placeholder={placeholder}
        onChange={event => onChange(event.target.value)}
      />
      <br />
      <span className="error">{errorMsg}</span>
    </div>
  );
};
