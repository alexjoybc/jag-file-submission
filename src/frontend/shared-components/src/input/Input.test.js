import React from "react";
import { Input } from "./Input";
import testBasicSnapshot from "../TestHelper";

describe("Input", () => {
  const input = {
    styling: "editable_white",
    id: "textInputId"
  };

  const onChange = jest.fn();

  test("Input field when mandatory and with label matches the snapshot", () => {
    const inputElement = (
      <Input
        input={{
          ...input,
          isRequired: true,
          placeholder: "Enter id",
          label: "Submission ID"
        }}
        onChange={onChange}
      />
    );

    testBasicSnapshot(inputElement);
  });

  test("Input field when not mandatory and with label matches the snapshot", () => {
    const inputElement = (
      <Input
        input={{
          ...input,
          isRequired: false,
          placeholder: "Enter id",
          label: "Submission ID"
        }}
        onChange={onChange}
      />
    );

    testBasicSnapshot(inputElement);
  });

  test("Input field when mandatory and no label matches the snapshot", () => {
    const inputElement = (
      <Input
        input={{
          ...input,
          isRequired: true,
          placeholder: "Enter id"
        }}
        onChange={onChange}
      />
    );

    testBasicSnapshot(inputElement);
  });

  test("Input field when not mandatory and no label matches the snapshot", () => {
    const inputElement = (
      <Input
        input={{
          ...input,
          isRequired: false,
          placeholder: "Enter id"
        }}
        onChange={onChange}
      />
    );

    testBasicSnapshot(inputElement);
  });
});
