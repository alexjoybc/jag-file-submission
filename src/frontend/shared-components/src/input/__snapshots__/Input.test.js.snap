// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Input Input field when mandatory and no label matches the snapshot 1`] = `
<div>
  <input
    className="editable_white"
    id="textInputId"
    onChange={[Function]}
    placeholder="Enter id"
    type="text"
  />
  <br />
  <span
    className="error"
  />
</div>
`;

exports[`Input Input field when mandatory and with label matches the snapshot 1`] = `
<div>
  <label
    className="label"
    htmlFor="textInputId"
  >
    <div
      className="textinput_label"
    >
      Submission ID
    </div>
    <span
      className="mandatory"
      id="asterisk"
    >
      *
    </span>
     
    <span
      className="note"
    />
  </label>
  <input
    className="editable_white"
    id="textInputId"
    onChange={[Function]}
    placeholder="Enter id"
    type="text"
  />
  <br />
  <span
    className="error"
  />
</div>
`;

exports[`Input Input field when not mandatory and no label matches the snapshot 1`] = `
<div>
  <input
    className="editable_white"
    id="textInputId"
    onChange={[Function]}
    placeholder="Enter id"
    type="text"
  />
  <br />
  <span
    className="error"
  />
</div>
`;

exports[`Input Input field when not mandatory and with label matches the snapshot 1`] = `
<div>
  <label
    className="label"
    htmlFor="textInputId"
  >
    <div
      className="textinput_label"
    >
      Submission ID
    </div>
    
     
    <span
      className="note"
    />
  </label>
  <input
    className="editable_white"
    id="textInputId"
    onChange={[Function]}
    placeholder="Enter id"
    type="text"
  />
  <br />
  <span
    className="error"
  />
</div>
`;
